src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -143,6 +143,149 @@ public abstract class Avx : Sse42
         /// </summary>
         public static Vector256<double> Compare(Vector256<double> left, Vector256<double> right, FloatComparisonMode mode) => Compare(left, right, mode);
 
+        /// <summary>
+        /// __m256 _mm256_cmpeq_ps (__m256 a,  __m256 b)
+        ///   CMPPS ymm, ymm/m256, imm8(0)
+        /// The above native signature does not exist. We provide this additional overload for completeness.
+        /// </summary>
+        public static Vector256<float> CompareEqual(Vector256<float> left, Vector256<float> right) => Compare(left, right, FloatComparisonMode.OrderedEqualNonSignaling);
```

Comment:
@tannergooding Can you explain why CompareEqual corresponds to OrderedEqual**Non**Signaling while CompareGreaterThan  corresponds to OrderedGreaterThanSignaling?

Reply:
The simplest explanation is because that is what the immediate values available to Sse/Sse2 are: 
![image](https://user-images.githubusercontent.com/10487869/77686315-fd42b000-6f59-11ea-8238-3bf5db263660.png)

The lengthier explanation is that `Signaling` when floating-point exceptions are supported (we don't support and explicitly disable them in .NET) throw for `SNaN` inputs but not for `QNaN`. The comparison operations like `==`, `!=`, `>`, `>=`, `<`, and `<=` don't expect `NaN` as inputs and so they signal by default; while the `Ordered`/`Unordered` comparisons do expect `NaN` inputs and so they don't.